test/test_runner_battle.c

```diff
@@ -650,8 +650,53 @@ static u32 BattleTest_RandomUniform(enum RandomTag tag, u32 lo, u32 hi, bool32 (
         }
     }
     //trials
-    if (tag == STATE->rngTag)
-        return RandomUniformTrials(tag, lo, hi, reject, caller);
+    switch (tag)
+    {
+        case RNG_AI_SCORE_TIE_SINGLES:
+        case RNG_AI_SCORE_TIE_DOUBLES_MOVE:
+            switch (DATA.scoreTieResolution)
+            {
+                case SCORE_TIE_HI:
+                    return (DATA.trial.scoreTieCount - 1);
+                case SCORE_TIE_RANDOM:
+                    if (DATA.trial.scoreTieCount == 0)
+                    {
+                        return 0; // Failsafe
+                    }
+                    else
+                    {
+                        u32 result = RandomUniformTrials(tag, lo, hi, reject, caller);
+                        return result;
+                    }
```

Comment:
```suggestion
                        return 0; // Failsafe
                    else
                        return RandomUniformTrials(tag, lo, hi, reject, caller);
```
No need to use an intermediary value here.